### PR TITLE
Delete OctoPrint logfiles configured via pattern and age

### PR DESCRIPTION
**Pull request by Reperiel**
I always have serial.log enabled. These use by large the most space in .octoprint/logs therefore I have created a crontab to delete old logfiles. There are two crontab entries created, one for boot and one daily. The boot entry is meant for users who are shutting their Raspi down between prints.

The pattern(s) (blank separated) and the age are configured via src/modules/octopi/config.

### CODE REVIEW

File: src/modules/octopi/start_chroot_script [hunk 1]
```diff
@@ -142,6 +142,12 @@ EOT
   # fetch current yq build and install to /usr/local/bin
   wget -O yq $OCTOPI_YQ_DOWNLOAD && chmod +x yq && mv yq /usr/local/bin
   
+  if [ -n "$OCTOPI_LOG_CLEAN_PATTERNS" -a -n "$OCTOPI_LOG_CLEAN_DAYS" ]; then
+    cmd="find \$HOME/.octoprint/logs -mtime +$OCTOPI_LOG_CLEAN_DAYS \\( "
+    names="$(and=; printf '%s\n' "$OCTOPI_LOG_CLEAN_PATTERNS" | tr ' ' '\n' | while read -r p; do [ -n "$p" ] || continue; printf "%s-name '%s'" "$and" "$p"; and=" -a "; done)"
+    cmd="$cmd$names \) -delete"
+    printf '@reboot %s\n@daily %s\n' "$cmd" "$cmd" | sudo -u pi crontab
```
Comment (tedder) [inline src/modules/octopi/start_chroot_script:149]: it'd be better to print/echo/cat this to a new file in /etc/cron.d/ or /etc/cron.daily/ if you don't mind losing the "reboot". Their syntaxes are slightly different, but it's better than blindly putting it into the UI-oriented crontab.

Comment (Reperiel) [reply src/modules/octopi/start_chroot_script:149]: /etc/cron.daily/ was my first thought but this would be run as root. On the other hand crontab(1) is the documented way to manage crontabs.

Quote from the man-page:
>       crontab  is  the program used to install, deinstall or list the tables used to
       drive the cron(8) daemon in Vixie Cron.  Each user can have their own crontab,
       and  though these are files in /var/spool/cron/crontabs, they are not intended
       to be edited directly.

**Comment by guysoft** [comment]
@foosel Do you approve of this? or shouldn't this be managed by OctoPrint?

**Comment by foosel** [comment]
Very hard corner case. Considering that it is not recommend to leave `serial.log` enabled at all times, and nothing should mess with the log files IF it is in fact enabled (because otherwise I can already see people generating a requested `serial.log`, then not getting around to sharing it for a couple of days, and then the thing that is desperately needed for debugging something is just gone) I'm strongly against merging this.

OctoPrint will already delete old files by itself. It only keeps the 3 old `serial.log`s, which are recreated on each connect, and `octoprint.log` will also be rotated daily and 6 old ones are kept.

Long story short: Please do not mess with the log files, especially not with an outside cronjob. If that is to be included, it needs a setting in `octopi.txt` that should default to off. Also don't leave `serial.log` enabled at all times.

**Comment by guysoft** [comment]
@Reperiel heyx sorry it seems that this won't make sense emerging in for everyone.
I hate to see functional code not getting in, but @foosel has a point.

If you really need this in multiple instances I could ship a script in the ``scripts`` folder that enableds it, but it won't be enabled by default for everyone.

**Comment by Reperiel** [comment]
@guysoft no problem. It was only that I had some large serial.log files and had therefore made my crontab entries. The Raspi 4 has more than enough power to leave this log always enabled so I thought others with less linux experience could get the benefit of the cleanup.

I had seen the serial.log didn't get the same rotation as octoprint.log and one of them was nearly 1GB.

As there is a reliable cleanup as @foosel said, there is not problem and my cron jobs are unnecessary anyway.

File: src/modules/octopi/start_chroot_script (hunk 1 above)
Comment (tedder) [reply src/modules/octopi/start_chroot_script:149]: I didn't say /var/spool; If you didn't want to use root, /etc/cron.d/ uses a user specifier, eg:
```
10 3 * * * pi echo "hello from pi user"
```

The reason I know is because I forget its syntax is different :/ (and yes, this is a moot point anyhow)

**Comment by tedder** [comment]
Yeah, the rotation on serial.log is .. different. I think it's after every process restart? Or every connect to the ttyusb? I dug into it once to do some better rotation and gave up when I realized it wasn't easily triggerable.
